### PR TITLE
Use CNS to bootstrap the Consul cluster

### LINKED ISSUE #17: Consul will fail to start up when multiple RFC1918 IPs are attached to the instance
For example, if you are running Consul in a network with multiple private IPs associated with NICs on the Consul instance, when Consul boots up, it will be unable to auto-select the network it should advertise on.

When using Consul on multiple private networks without Docker, this is a simple fix - you just specify the IP of the private network that you want to advertise on. However, when you use Docker, specifying the IP as part of the startup parameters gets you into a chicken and egg problem.

Ideally, ContainerPilot could support templating in the available IPs into the run command executed.

### PR DESCRIPTION
**Pull request by tgross**
For https://github.com/autopilotpattern/consul/issues/14. Also fixes https://github.com/autopilotpattern/consul/issues/17, does the cleanup in https://github.com/autopilotpattern/consul/issues/13, and obviates https://github.com/autopilotpattern/consul/issues/16

A node's health check will check to see if it is a member of a raft. If it has not peers then it will try to join the raft via the CNS svc record. Once a node is connected to at least one other peer it'll get the rest of the raft via the Consul LAN gossip. If a node resolves itself in CNS then we just end up retrying on the next health check until we've interconnected all the nodes and gossiped membership.

TODO:
- [x] update the README
- [x] make a decision on which of the two gross hacks in the local-compose.yml file to use; we don't have DNS unless we set up a user-defined network in Docker. Alternately, we can insist on not having a hack and make the end-user set up a user-defined network for local testing. I'm open to suggestion here.

cc @misterbisson @dekobon

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -25,22 +22,27 @@ RUN curl -Lo /tmp/webui.zip https://releases.hashicorp.com/consul/${CONSUL_VERSI
...
+    && echo "${CONTAINERPILOT_CHECKSUM}  /tmp/containerpilot.tar.gz" | sha1sum -c \
+    && tar zxf /tmp/containerpilot.tar.gz -C /usr/local/bin \
+    && rm /tmp/containerpilot.tar.gz
 
 # Consul config
-COPY ./config /config/
-ONBUILD ADD ./config /config/
+COPY etc/containerpilot.json etc/
+COPY etc/consul.json etc/consul/
+#ONBUILD ADD ./config /config/
```
Comment (tgross) [inline Dockerfile]: What was the ONBUILD for here, @misterbisson?

Comment (misterbisson) [inline Dockerfile]: um... I don't recall.

Comment (misterbisson) [inline Dockerfile]: Let's cut it

**Comment by misterbisson** [comment]
> make a decision on which of the two gross hacks in the local-compose.yml file to use; we don't have DNS unless we set up a user-defined network in Docker. Alternately, we can insist on not having a hack and make the end-user set up a user-defined network for local testing

I'd lean to the less hacky approach, especially since it's inline with overall Docker practice. It'll be more work, but at least it's better documented and understood than a one-off hack.

**Comment by tgross** [comment]
> I'd lean to the less hacky approach, especially since it's inline with overall Docker practice. It'll be more work, but at least it's better documented and understood than a one-off hack.

Aye, will do.

**Comment by tgross** [comment]
I've updated the README and made the changes to the local-compose.yml file, @misterbisson

**Comment by misterbisson** [comment]
🏡 🚶

**Pull request merged**